### PR TITLE
[RF] Fix ROOT-10668.

### PR DESCRIPTION
**Pull request by hageboeck**
When asymptotically correct errors are used, the current implementation
breaks if variables don't have identical name and title. This is solved
by always using the name.

### CODE REVIEW

File: roofit/roofitcore/src/RooAbsPdf.cxx [hunk 1]
```diff
@@ -1508,9 +1508,10 @@ RooFitResult* RooAbsPdf::fitTo(RooAbsData& data, const RooLinkedList& cmdList)
 	   *obs = *data.get(j);
 	   //Determine first derivatives
 	   std::vector<Double_t> diffs(floated.getSize(), 0.0);
-	   for (int k=0; k<floated.getSize(); k++) {
-	      RooRealVar* paramresult = (RooRealVar*)floated.at(k);
-	      RooRealVar* paraminternal = (RooRealVar*)floatingparams->find(paramresult->getTitle());
+	   for (int k=0; k < floated.getSize(); k++) {
+	      auto paramresult = dynamic_cast<const RooRealVar*>(floated.at(k));
```
Comment (eguiraud) [inline roofit/roofitcore/src/RooAbsPdf.cxx:1512]: are you adding const-ness through a dynamic cast? `const auto *paramresult = dynamic/static_cast<RooRealVar*>` might be less surprising

File: roofit/roofitcore/src/RooAbsPdf.cxx [hunk 2]
```diff
@@ -1495,9 +1495,9 @@ RooFitResult* RooAbsPdf::fitTo(RooAbsData& data, const RooLinkedList& cmdList)
 	std::vector<std::unique_ptr<RooDerivative> > derivatives;
 	const RooArgList& floated = rw->floatParsFinal();
 	std::unique_ptr<RooArgSet> floatingparams( (RooArgSet*)getParameters(data)->selectByAttrib("Constant", false) );
-	for (int k=0; k<floated.getSize(); k++) {	   
-	   RooRealVar* paramresult = (RooRealVar*)floated.at(k);
-	   RooRealVar* paraminternal = (RooRealVar*)floatingparams->find(paramresult->getTitle());
+	for (const auto paramresult : floated) {
+	   auto paraminternal = dynamic_cast<RooRealVar*>(floatingparams->find(*paramresult));
+	   assert(paraminternal);
```
[the remark this answers is not in the document]
Comment (hageboeck) [reply roofit/roofitcore/src/RooAbsPdf.cxx:1500]: That's actually correct. By design, parameters can only be of type `RooRealVar`.

[the remark this answers is not in the document]
Comment (eguiraud) [reply roofit/roofitcore/src/RooAbsPdf.cxx:1500]: cool, but i'm getting mixed signals then :smile: why is the conversation resolved? :sweat_smile:

[the remark this answers is not in the document]
Comment (eguiraud) [reply roofit/roofitcore/src/RooAbsPdf.cxx]: ok now resolved :D

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by hageboeck** [comment]
@pcanal What's that thing about not excluding tests? Should that be our new default? Why isn't this the default?

**Pull request merged**